linkerd/app/outbound/src/tests/tcp.rs

```diff
@@ -402,6 +599,48 @@ async fn no_profiles_when_outside_search_nets() {
     );
 }
 
+struct Endpoint {
+    identity: tls::Conditional<linkerd2_identity::Name>,
+    count: Arc<AtomicUsize>,
+    enabled: Arc<AtomicBool>,
+}
+
+impl Default for Endpoint {
+    fn default() -> Self {
+        Self {
+            identity: tls::Conditional::None(
+                tls::ReasonForNoPeerName::NotProvidedByServiceDiscovery,
+            ),
+            count: Arc::new(AtomicUsize::new(0)),
+            enabled: Arc::new(AtomicBool::new(true)),
+        }
+    }
+}
+
+impl Into<Box<dyn FnMut(TcpEndpoint) -> test_support::connect::ConnectFuture + Send + 'static>>
```

Comment:
tioli. This strikes me as pretty subtle. It might be a little clearer as an explicitly named fn like `into_connect`? Not a blocker